processing/src/main/java/io/druid/query/groupby/having/DimensionSelectorHavingSpec.java

```diff
@@ -48,11 +48,7 @@ public DimensionSelectorHavingSpec(
   {
     dimension = Preconditions.checkNotNull(dimName, "Must have attribute 'dimension'");
     this.value = value;
-    if (extractionFn != null) {
-      this.extractionFn = extractionFn;
-    } else {
-      this.extractionFn = new IdentityExtractionFn(); 
-    }
+    this.extractionFn = extractionFn != null ? extractionFn : IdentityExtractionFn.getInstance();
```

Comment:
you mean remove getInstance() method and make "instance" public so that callers can just get reference to that?
getters are considered a good practice in general, so I have put the getter probably out of habit, do you think extra method call in this case is going to introduce any significant performance difference?